src/Symfony/Component/EventDispatcher/DependencyInjection/InvalidBeforeAfterListenerDefinitionException.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\EventDispatcher\DependencyInjection;
+
+use Symfony\Component\DependencyInjection\Exception\InvalidArgumentException;
+
+/**
+ * @psalm-import-type BeforeAfterDefinition from RegisterListenersPass
+ */
+final class InvalidBeforeAfterListenerDefinitionException extends InvalidArgumentException
+{
+    private function __construct(string $errorServiceId, string $message)
+    {
+        parent::__construct(sprintf('Invalid before/after definition for service "%s": %s', $errorServiceId, $message));
+    }
+
+    public static function beforeAndAfterAtSameTime(string $errorServiceId): self
```

Comment:
We generally don't use named constructor for exceptions in Symfony. Build the exception message in the compiler pass in the place triggering the error.

I'm not even sure we need a dedicated exception class. `Symfony\Component\DependencyInjection\Exception\InvalidArgumentException` could be used directly.

Reply:
done, but I feel like it is a bad move and makes the code less readable